core/src/conn.rs

```diff
@@ -0,0 +1,35 @@
+use mongodb::sync::Client;
+use std::error::Error;
+use std::time::Duration;
+
+#[derive(Debug)]
+pub struct MongoConnection {
+    // The mongo DB client
+    pub client: Client,
+    // The current database set for this client.
+    // All new queries will be done on this DB.
+    pub current_db: Box<str>,
```

Comment:
```suggestion
    pub current_db: String,
```

Reply:
`Box<str>` is effectively `String`.

Reply:
https://users.rust-lang.org/t/use-case-for-box-str-and-string/8295

`Box<str>` is _slightly_ smaller, if we never want to append to `current_db`, so I'll leave this up to you, depending on how much we care about a few bytes.

Reply:
As discussed during standup, I'll replace all Box<str> by String.

Reply:
Done.